features/nvstore/source/nvstore_shared_lock.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2016 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+
+#ifndef __NVSTORE_OS_WRAPPER_H
+#define __NVSTORE_OS_WRAPPER_H
+
+#include <stdint.h>
+#include "rtos/Mutex.h"
+
+enum {
+    NVSTORE_OS_OK          =  0,
+    NVSTORE_OS_RTOS_ERR    = -1,
+    NVSTORE_OS_INV_ARG_ERR = -2,
+    NVSTORE_OS_NO_MEM_ERR  = -3,
+};
+
+class SharedLock {
```

Comment:
I am interested in this SharedLock - missing docs here to describe the scope of this. And if we do not have already something similar in the codebase?

Reply:
This definitely needs documentation, which will be added. This class implements a shared lock mechanism (AKA RW lock). Now this implementation isn't like the classic one ,the one that needs two Mutexes and a counter. Instead it has one Mutex and an atomic counter, favoring the usage of shared lock/unlock on the count of the exclusive lock/unlock. This will be added in the documentation.

Reply:
And there doesn't seem to be an implementation for it, at least not under rtos.